src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -285,32 +285,30 @@ private DataAccessException getFaultToInsert(LockType lockType) {
     @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
     public UUID createAndLock(Dataset dataset, String flightId) throws IOException, SQLException {
         logger.debug("Lock Operation: createAndLock datasetId: {} for flightId: {}", dataset.getId(), flightId);
+        Timestamp created_date = new Timestamp(dataset.getCreatedDate().toEpochMilli());
```

Comment:
There is nothing wrong with this, but FYI you can use DaoKeyHolder to get any column data from the inserted row, not just the id.

Reply:
I think i did not understand the first time i looked at this, but i do not think we want our users to have to provide the created data input. I think we should still allow the database to generate that as the default value and then retrieve it using the key holder. Sorry did not understand this the first time around.